albumentations/augmentations/transforms.py

```diff
@@ -3550,3 +3553,61 @@ def apply(self, img, transforms=(), **params):
 
     def get_transform_init_args_names(self):
         return ("brightness", "contrast", "saturation", "hue")
+
+
+class Superpixels(ImageOnlyTransform):
+    """Completely or partially transform the input image to its superpixel representation. Uses skimage's version
+    of the SLIC algorithm. May be slow.
+    Args:
+        p_replace (float): defines the probability of any superpixel area being replaced by the superpixel, i.e. by
+            the average pixel color within its area. Default: 0.1.
+        n_segments (int): target number of superpixels to generate. Default: 100.
+        p (float): probability of applying the transform. Default: 0.5.
+    Targets:
+        image
+    """
+
+    def __init__(self, p_replace=0.1, n_segments=100, always_apply=False, p=0.5):
+        super(Superpixels, self).__init__(always_apply, p)
+        self.p_replace = p_replace
+        self.n_segments = n_segments
+
+    def apply(self, image, cols, rows):
+        segments = slic(image, n_segments=self.n_segments, compactness=5)
```

Comment:
`n_segments` must be selected from the range.

Reply:
Also maybe would be great to provide `compactness` as a transform argument.
In the latest imgaug version default value 10.